### PR TITLE
Bundle

### PR DESCRIPTION
**Pull request by GuillaumeRoeder**

### CODE REVIEW

File: silex_client/commands/copy.py [hunk 1]
```diff
@@ -135,6 +136,10 @@ async def __call__(
 
                 await execute_in_thread(self.copy, src_path, dst_path, progress)
 
+                # Check if copy was done successfully
+                if not os.path.exists(dst_path):
+                    Exception(f"Fail to copy {src_path} to {dst_path}")
```
Comment (Acedyn) [inline silex_client/commands/copy.py]: forget raise

File: silex_client/commands/build_bundle_path.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        },
+         "frame_set": {
+            "type": fileseq.FrameSet,
+            "value": fileseq.FrameSet(0),
+            "hide": True,
+        },
+        "padding": {
+            "label": "padding for index in sequences",
+            "type": int,
+            "value": 1,
```
Comment (Acedyn) [inline silex_client/commands/build_bundle_path.py:41]: true

File: silex_client/commands/expand_paths.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        action_query: ActionQuery,
+        logger: logging.Logger,
+    ):
+
+        file_paths = parameters['paths_to_expand']
+
+        expanded_paths = list(map(expand_environement_variable,file_paths))
+        environement_var = list(set([find_environement_variable(path).group(1) for path in file_paths]  )) 
+
+        return {"expanded_paths": expanded_paths , 'envs': environement_var}
```
Comment (Acedyn) [inline silex_client/commands/expand_paths.py]: environment

File: silex_client/commands/set_env.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+from silex_client.utils.files import expand_environement_variable
+from silex_client.utils.parameter_types import ListParameter
+
+if typing.TYPE_CHECKING:
+    from silex_client.action.action_query import ActionQuery
+
+
+class SetEnv(CommandBase):
+    """
+    Replace path by environement variables if it exists
```
Comment (Acedyn) [inline silex_client/commands/set_env.py]: environment

File: silex_client/commands/rename.py [hunk 5]
```diff
@@ -21,6 +21,7 @@
     TextParameterMeta,
 )
 from silex_client.utils.thread import execute_in_thread
+from silex_client.utils import files
```
Comment (Acedyn) [inline silex_client/commands/rename.py]: check if  impoorted

File: silex_client/commands/select_bundle.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+from silex_client.action.command_base import CommandBase
+from silex_client.resolve.config import Config
+from silex_client.utils.parameter_types import PathParameterMeta
+
+# Forward references
+if typing.TYPE_CHECKING:
+    from silex_client.action.action_query import ActionQuery
+
+
+class SelectBundle(CommandBase):
```
Comment (Acedyn) [inline silex_client/commands/select_bundle.py]: checker heritage

File: silex_client/commands/select_bundle_reference.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+from silex_client.action.command_base import CommandBase
+from silex_client.resolve.config import Config
+from silex_client.utils.parameter_types import PathParameterMeta
+
+# Forward references
+if typing.TYPE_CHECKING:
+    from silex_client.action.action_query import ActionQuery
+
+
+class SelectBundleReference(CommandBase):
```
Comment (Acedyn) [inline silex_client/commands/select_bundle_reference.py]: checker heritage

File: silex_client/utils/constants.py [hunk 8]
```diff
@@ -11,3 +11,10 @@
     re.compile(r"^.+[^\w#](#+)\W.+$"),
 ]
 
+# Match environement variable in path for differente format
```
Comment (Acedyn) [inline silex_client/utils/constants.py]: environment
different

File: silex_client/utils/files.py [hunk 9]
```diff
@@ -20,6 +22,56 @@
 ERROR_INVALID_NAME = 123
 
 
+def reload_recursive(parent_module: ModuleType) -> None:
```
Comment (Acedyn) [inline silex_client/utils/files.py]: delete, no use anymore

**Review by Acedyn — CHANGES_REQUESTED** [review]

**Review by johhnry — APPROVED** [review]
LGTM

**Review by Acedyn — APPROVED** [review]

**Pull request merged**